docs/development.md

```diff
@@ -0,0 +1,138 @@
+# Development
+
+## Build with Docker
+
+To run the entire build process for both TypeScript and Python: \
+Ensure you have Docker installed and you are in the root directory of gink.
+
+```sh
+docker build .
+```
+
+This builds everything and runs tests and linters. This is what is run on push to GitHub.
+
+## Build without Docker
+
+This project uses a Makefile to handle most of the building processes. Before we jump in, ensure you have make, protobuf-compiler, and curl installed. If not, run the following commands:
+
+### Prerequisites
+
+```sh
+apt-get install make protobuf-compiler curl -y
+```
+
+This process assumes you have npm, pip, and python venv installed.
+
+```sh
+apt-get install npm python3-pip python3-venv -y
+```
+
+### Install dependencies
+
+```sh
+make install-debian-packages && \
```

Comment:
So this might be a bit confusing.  If all the Debian packages listed in packages.txt are installed then there's no need to install additional python packages.